java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzChangeEventConsumer.java

```diff
@@ -168,14 +181,66 @@ var record = event.value();
                                         record.topic(), record.valueSchema(), record.value());
                         var message =
                                 msgBuilder
-                                        .setIsTransactionMeta(true)
+                                        .setMsgType(CdcMessage.CdcMessageType.TRANSACTION_META)
                                         .setPayload(new String(payload, StandardCharsets.UTF_8))
                                         .setSourceTsMs(trxTs)
                                         .build();
                         LOG.debug("transaction => {}", message);
                         respBuilder.addEvents(message);
                         break;
                     }
+
+                case SCHEMA_CHANGE:
+                    {
+                        var ddl = ((Struct) record.value()).getString("ddl");
+                        if (ddl.contains("CREATE") || ddl.contains("DROP")) {
+                            LOG.info("skip create/drop table event");
+                            continue;
+                        }
+
+                        var sourceStruct = ((Struct) record.value()).getStruct("source");
+                        if (sourceStruct == null) {
+                            throw new CdcConnectorException(
+                                    "source field is missing in schema change event");
+                        }
+
+                        // upstream event time
+                        long sourceTsMs = sourceStruct.getInt64("ts_ms");
+                        // concat full table name, right now we only support MySQL schema change
+                        // event
+                        var fullTableName =
+                                String.format(
+                                        "%s.%s",
+                                        sourceStruct.getString("db"),
+                                        sourceStruct.getString("table"));
```

Comment:
Should we use the `id` under `tableChanges` instead?

Reply:
removed the extraction of `fullTableName`